src/components/TableNew/TablePager.tsx

```diff
@@ -0,0 +1,87 @@
+import { Button } from '@/components/ButtonNew/Button'
+import { Paragraph } from '@/components/TypographyNew'
+import { useEffect, useState } from 'react'
+
+interface TablePagerProps {
+  pageSize: number
+  totalItems: number
+  pagedItemName: string
+  onPageChange: (startIndex: number, endIndex: number) => void
+  mode: 'cyclic' | 'expandable'
+}
+
+export const TablePager: React.FC<TablePagerProps> = ({
+  pageSize,
+  totalItems,
+  pagedItemName,
+  onPageChange,
+  mode,
+}) => {
+  const [currentPage, setCurrentPage] = useState(0)
+  const [expandEnd, setExpandEnd] = useState(pageSize)
+
+  let start: number
+  let end: number
+  let isButtonDisabled = false
+
+  if (mode === 'expandable') {
+    start = totalItems === 0 ? 0 : 1
+    end = Math.min(expandEnd, totalItems)
+    isButtonDisabled = end >= totalItems
+  } else {
+    start = totalItems === 0 ? 0 : currentPage * pageSize + 1
+    end = Math.min((currentPage + 1) * pageSize, totalItems)
+    // Clamp start and end to never exceed totalItems
+    if (start > totalItems) start = totalItems
+    if (end < start) end = start
+  }
+
+  const hasItems = totalItems > 0
+
+  const handleNext = () => {
+    if (mode === 'expandable') {
+      const newEnd = Math.min(expandEnd + pageSize, totalItems)
```

Comment:
```suggestion
  const handleNext = () => {
  if (totalItems === 0) return
    if (mode === 'expandable') {
      const newEnd = Math.min(expandEnd + pageSize, totalItems)
```
I would include that to avoid unexpected behaviors.